### PR TITLE
[TR] PIM-4018: Highlight modified fields in product draft mode

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-4018

| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | Y
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | -
| Scenarios pass?      | -
| Checkstyle issues?*  | N
| PMD issues?**        | -
| Changelog updated?   | N
| Fixed tickets        | -
| DB schema updated?   | N
| Migration script?    | N
| Doc PR               |

EE PR : https://github.com/akeneo/pim-enterprise-dev/pull/893

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 1]
```diff
@@ -175,6 +180,18 @@ define([
 
                 productValue.value = value;
                 mediator.trigger('entity:form:edit:update_state');
+            },
+            addHtmlClass: function (className) {
+                if (-1 === this.htmlClasses.indexOf(className)) {
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: `_.has()` could be use no?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes.js [hunk 2]
```diff
@@ -59,8 +59,8 @@ define(
 
                 this.listenTo(this.getRoot().model, 'change', this.render);
                 this.listenTo(UserContext, 'change:catalogLocale change:catalogScope', this.render);
+                this.listenTo(mediator, 'entity:action:validation_error', this.render);
                 mediator.on('product:action:post_update', _.bind(this.postSave, this));
-                mediator.on('product:action:post_validation_error', _.bind(this.postValidationError, this));
                 mediator.on('show_attribute', _.bind(this.showAttribute, this));
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes.js]: `show_attribute` maybe too generaic for an event name.

**Comment by grena** [comment]
GTM for me if my comment makes no sense :smile:

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes.js (hunk 2 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes.js]: Agree, we should clean events management in another PR.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/attribute-group-selector.js [hunk 3]
```diff
@@ -33,8 +34,39 @@ define(
                 this.listenTo(this.state, 'change', this.render);
                 this.badges = {};
 
+                this.stopListening(mediator, 'entity:action:validation_error');
+                this.listenTo(mediator, 'entity:action:validation_error', this.onValidationError);
+
+                this.stopListening(mediator, 'product:action:post_update');
+                this.listenTo(mediator, 'product:action:post_update', this.onPostUpdate);
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/attribute-group-selector.js]: Why `this.stopListening` before `this.listenTo` in the constructor ?
The instance should not exist before, and so should not listen to anything.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: It works on keys of an object, here i'm just working on a simple array of strings. ; )

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/attribute-group-selector.js (hunk 3 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/attribute-group-selector.js]: Seems weird to me too, same comment as above.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 4]
```diff
@@ -35,20 +35,24 @@ define([
...
+                this.htmlClasses = [];
 
                 return this;
             },
             render: function () {
                 this.setEditable(true);
-                var promises = [];
+                this.setValid(true);
+                var promises  = [];
+                this.elements = {};
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: invert this two lines (group `this.` affectation together :))

**Comment by sumbobyboys** [comment]
GTM after fixes

**Pull request merged**